runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,357 @@
+                    }
+
+                    txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinations, null);
+                    txn.commit();
+
+                    return null;
+                } catch (TransactionAbortedException tae) {
+                    log.error("Unable to add destinations, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Unable to add destinations");
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Remove destination from LogReplicationSourceMetadataTable
+     *
+     * @param logicalGroup String representative of the group of tables which are replicated
+     * @param destination The cluster from which the logicalGroup should be removed from all future replication
+     */
+    public void removeDestination(String logicalGroup, String destination) {
+        Preconditions.checkArgument(isValid(logicalGroup), "logicalGroup is null or empty");
+        Preconditions.checkArgument(isValid(destination), "destination is null or empty");
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = ClientDestinationInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        DestinationInfoVal clientDestinationIds = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+                        List<String> clientDestinationIdsList = new ArrayList<>(clientDestinationIds.getDestinationIdsList());
+
+                        if (clientDestinationIdsList.remove(destination)) {
+                            if (clientDestinationIdsList.size() == 0) {
+                                txn.delete(sourceMetadataTable, clientInfoKey);
+                            } else {
+                                clientDestinationIds = clientDestinationIds.toBuilder()
+                                        .clearDestinationIds()
+                                        .addAllDestinationIds(clientDestinationIdsList)
+                                        .build();
+                                txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinationIds, null);
+                            }
+                            txn.commit();
+
+                            return null;
+                        }
+                        throw new NoSuchElementException("No matching destination found");
```

Comment:
here too, the operation must not fail IMO.  We can log a debug message and return